kube-core/src/request.rs

```diff
@@ -148,9 +148,35 @@ impl Request {
             qp.append_pair("labelSelector", labels);
         }
         let urlstr = qp.finish();
-        let body = serde_json::to_vec(&dp).map_err(Error::SerializeBody)?;
+
+        let data = if dp.is_default() {
+            vec![] // default serialize needs to be empty body
+        } else if std::env::var("KUBE_UNSTABLE_V1_25_DELETE").is_ok() {
+            // Empty delete params triggers an error in kubernetes v1.25
+            // https://github.com/kubernetes/kubernetes/issues/111985
```

Comment:
how about
```
            // The kube-apiserver in Kubernetes v1.25.0 has broken deserialization of non-default delete options
            // So we have to wrap our DeleteParams inside a kind/apiVersion struct to workaround it
            // https://github.com/kubernetes/kubernetes/issues/111985
            // This workaround does not work on older Kubernetes versions so it has to be opt-in.

            // This is a hack that we plan on removing in the next kube release
            // when upstream hopefully resolves the issue.
```

Reply:
Personally, I would wait.

`kind 0.15.0` only has a 1.25.0 image and 1.25.1 is already out. Yes, it will be a while before a proper fix exists, but putting work-arounds in place would just add additional work.

Time box it (say `1.25.3`), and if not resolved - merge, and release.

Reply:
Yeah, that's much better :+1:
The bug is much worse than I thought.

Reply:
I think we want to release it because requiring users to use older `k8s_openapi` is often confusing.

I don't know how many users will be testing against 1.25 anytime soon, so this might not be worth including. This is convenient, but we can document it, and users can always fork and add it themselves to test if they'd like.

Reply:
Sure :+1: